src/Interpreters/ITokenExtractor.h

```diff
@@ -179,8 +194,43 @@ struct SplitTokenExtractor final : public ITokenExtractorHelper<SplitTokenExtrac
     void substringToBloomFilter(const char * data, size_t length, BloomFilter & bloom_filter, bool is_prefix, bool is_suffix) const override;
 
     void substringToGinFilter(const char * data, size_t length, GinFilter & gin_filter, bool is_prefix, bool is_suffix) const override;
+};
 
+/// Parser doing "no operation". Returns the entire input as a single token.
+struct NoOpTokenExtractor final : public ITokenExtractorHelper<NoOpTokenExtractor>
+{
+    static const char * getName() { return "noop"; }
+    static const char * getExternalName() { return getName(); }
```

Comment:
The tokenizer names in namespace `detail` were replaced by `getExternalName()`. We should keep classes and names in the same place.